src/object/tests/crush.test.ts

```diff
@@ -33,4 +33,36 @@ describe('crush function', () => {
       timestamp: now
     })
   })
+  test('handles property names with dots 1', () => {
+    const obj = {
+      a: { 'b.c': 'value' }
+    }
+    expect(_.crush(obj)).toEqual({
+      'a.b.c': 'value'
+    })
+  })
+  test('handles property names with dots 2', () => {
+    const obj = {
+      'a.b': { c: 'value' }
+    }
+    expect(_.crush(obj)).toEqual({
+      'a.b.c': 'value'
+    })
+  })
+  test('handles property names with dots 3', () => {
+    const obj = {
+      'a.b': { 'c.d': 123.4 }
+    }
+    expect(_.crush(obj)).toEqual({
+      'a.b.c.d': 123.4
+    })
```

Comment:
This doesn't preserve the intent though. We'll probably need support for `"a.b"."c.d"` paths.

Reply:
@aleclarson I don't understand your remark.
The test is just a slightly more elaborate example of key names with a dot inside.
How does this not preserve intent ?  It's just a test, it passes.
What do you mean with `"a.b"."c.d" paths` ?  AFAIK it's not possible to specify such a path in javascript.

Reply:
Sorry for the vagueness. We can still merge this PR, since you fixed `crush` to pass the failed test case.

But in the future, we should avoid crushing `["a.b"]["c.d"]` to `a.b.c.d` since that won't reproduce the same object when used with `construct`. Example below:

```ts
import { crush, construct } from 'radashi'

test('crush is the opposite of construct', () => {
  const source = { 'a.b': { 'c.d': 1 } }

  // This currently fails:
  expect(construct(crush(source))).toEqual(source)
})
```

So we can merge this PR now or, if you'd like, you could fix this test case as well. Up to you!

Reply:
There may be purposes for `crush()` other than being the inverse of `construct()`.
Probably a `get()` will be a good alternative for that in most cases, but the current `crush()` implementation is flawed as it returns `undefined` when there's dots in the propserty names..
I have made a new commit where the inner recursive function is outside of the `crush()` function,  That should be slightly more efficient in term of closures.

Reply:
True. Okay, awaiting your new commit. 👍

Reply:
@aleclarson 
I am unable to commit to my own PR ?
Get this error messages.  Any idea what might be going on ?
```
$ git push
remote: Permission to radashi-org/radashi.git denied to stefaanv.
fatal: unable to access 'https://github.com/radashi-org/radashi.git/': The requested URL returned error: 403
```

Reply:
I don't know what your `git remote -v` looks like, but seems like you need to set the branch's upstream with:

```
git remote set-url origin https://github.com/yourusername/radashi.git
git push -u origin your-branch-name
```

Replace 'yourusername' with your GitHub username and 'your-branch-name' with your actual branch name. This will update your origin to point to your fork and then push your branch there.

After this, you can just use `git push` for future pushes on this branch.

Reply:
@aleclarson thanks for the help, I succeeded in committing this changes.
You can merge the PR